### PR TITLE
BREAKING CHANGE: React Field component style props split

### PR DESCRIPTION
**Pull request by dbanksdesign**
#### Description of changes

* No longer splitting out Flex and Base style props on Field primitives. Instead, removing all style props to be applied to the wrapper element.
* Adding `inputStyles` prop where developers can send style props to the input element (or textarea, select)
* For TextAreaField destructuring the `resize` style prop because it makes sense to apply that to the textarea element.

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react/src/primitives/shared/styleUtils.ts [hunk 1]
```diff
@@ -158,124 +159,32 @@ export const useStyles = (
   );
 };
 
-/**
- * Map of all the FlexContainerStyleProps type keys
- * The type requires all keys in order to ensure it remains
- * in sync with the FlexContainerStyleProps type.
- */
-const FlexContainerStylePropsMap: Required<{
```
Comment (reesscot) [inline packages/react/src/primitives/shared/styleUtils.ts:166]: Nice, this simplifies our style types significantly!

**Review by reesscot — COMMENTED** [review]
LGTM, this really simplifies our style prop handling 💯

**Review by reesscot — DISMISSED** [review]
LGTM

File: packages/react/src/primitives/shared/__tests__/styleUtils.test.tsx [hunk 2]
```diff
@@ -463,15 +464,13 @@ describe('useTransformStyleProps', () => {
 
 describe('splitPrimitiveProps', () => {
   it('should split props into base, flex and rest', () => {
```
Comment (zchenwei) [inline packages/react/src/primitives/shared/__tests__/styleUtils.test.tsx:466]: We also want to change the test name?

File: docs/src/pages/[platform]/components/switchfield/examples/SwitchFieldStylePropsExample.tsx [hunk 3]
```diff
@@ -1,14 +1,9 @@
-import { SwitchField, useTheme } from '@aws-amplify/ui-react';
+import { SwitchField } from '@aws-amplify/ui-react';
 
 export const SwitchFieldStylePropsExample = () => {
-  const { tokens } = useTheme();
   return (
     <>
```
Comment (zchenwei) [inline docs/src/pages/[platform]/components/switchfield/examples/SwitchFieldStylePropsExample.tsx:5]: nit: we do not need this awkward `<></>` anymore

**Review by zchenwei — APPROVED** [review]

**Review by zchenwei — APPROVED** [review]
LGTM

**Review by reesscot — APPROVED** [review]

**Pull request merged**